go/throttle/throttler.go

```diff
@@ -375,6 +377,30 @@ func (throttler *Throttler) RecentAppsMap() (result map[string](*base.RecentApp)
 	return result
 }
 
+// markMetricHealthy will mark the time "now" as the last time a given metric was checked to be "OK"
+func (throttler *Throttler) markMetricHealthy(metricName string) {
+	throttler.metricsHealth.Set(metricName, time.Now(), cache.DefaultExpiration)
+}
+
+// timeSinceMetricHealthy returns time elapsed since the last time a metric checked "OK"
+func (throttler *Throttler) timeSinceMetricHealthy(metricName string) time.Duration {
+	if lastOKTime, found := throttler.metricsHealth.Get(metricName); found {
+		return time.Since(lastOKTime.(time.Time))
+	}
+	// Never been OK? This can happen; metric is bad ever since freno started running.
+	// There is no correct value to return here; we return a negative value to indicate "strange" or "Unknown"
+	return -time.Hour
```

Comment:
I have concerns about expressing `nil` semantics with a magic number. 

In [localCheck](https://github.com/github/freno/pull/44/files#diff-11634637460cb749624d521c95a9eaeeR112) we always set a value for the gauge indicating the metric, but only if a check has already been made first.

If I understand the code correctly, 1) not all metrics for existing backends will be available upfront (as a check needs to happen first to that said backend), and 2) As per https://github.com/rcrowley/go-metrics/blob/master/gauge.go#L58 a guage cannot contain nil values.

My recommendation will be that until the metric is healthy we don't include the metric in the result of `metrics-health` thus having least astonishing results for the api-call.

Please correct me if any of my assumptions was incorrect, or you have a different opinion. ❤️